### PR TITLE
fix: Marshal() including "default=" tags in EnvSet

### PR DESCRIPTION
**Pull request by kaandesu**
Hello, I experienced the same issue in #23 and in this PR attempted to fix it.

### CODE REVIEW

File: env.go [hunk 1]
```diff
@@ -327,6 +327,12 @@ func Marshal(v interface{}) (EnvSet, error) {
 		}
 
 		for _, envKey := range envKeys {
+			if strings.Contains(envKey, "=") {
+				switch strings.ToLower(strings.SplitN(envKey, "=", 2)[0]) {
+				case "separator", "required", "default":
```
Comment (patricksanders) [inline env.go:332]: Can you add a few test cases for `separator` and `required` as well?

File: env.go [hunk 2]
```diff
@@ -327,6 +327,12 @@ func Marshal(v interface{}) (EnvSet, error) {
 		}
 
 		for _, envKey := range envKeys {
+			if strings.Contains(envKey, "=") {
```
Comment (patricksanders) [inline env.go:330]: nit: this could use a comment to explain why these are being skipped

**Review by patricksanders — CHANGES_REQUESTED** [review]
Thanks for the PR. This looks good, just needs a few more test cases.

File: env.go (hunk 1 above)
Comment (kaandesu) [reply env.go:333]: Test cases for separator and required were added in the last commit 👍

**Pull request merged**